File: backend-java/src/main/java/com/aiinterview/model/Question.java
package com.aiinterview.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Table(name = "questions")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Question {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, columnDefinition = "TEXT")
    private String text;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private QuestionCategory category;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private QuestionDifficulty difficulty;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;
    
    @Column(name = "is_active")
    private Boolean isActive = true;
    
    @Column(columnDefinition = "TEXT")
    private String expectedAnswer;
    
    @Column(columnDefinition = "TEXT")
    private String keywords;  // JSON array of keywords
    
    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    public enum QuestionCategory {
        HR, TECHNICAL
    }
    
    public enum QuestionDifficulty {
        EASY, MEDIUM, HARD
    }
}

File: flask/Emotion_detection_with_CNN-main/models/question.py
from datetime import datetime
from extensions import db

class Question(db.Model):
    """Question model for interview question bank"""
    __tablename__ = 'questions'
    
    id = db.Column(db.Integer, primary_key=True)
    text = db.Column(db.Text, nullable=False)
    category = db.Column(db.String(20), nullable=False)  # 'hr' or 'technical'
    difficulty = db.Column(db.String(10), nullable=False, default='Medium')  # 'Easy', 'Medium', 'Hard'
    created_by = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    
    # Relationship to User (who created the question)
    creator = db.relationship('User', backref='created_questions')
    
    def to_dict(self):
        """Convert question to dictionary"""
        return {
            'id': self.id,
            'text': self.text,
            'category': self.category.lower(),
            'difficulty': self.difficulty,
            'created_by': self.created_by,
            'created_at': self.created_at.isoformat()
        }
    
    def __repr__(self):
        return f'<Question {self.id}: {self.category} - {self.difficulty}>'